cpp/ql/test/query-tests/Likely Bugs/Format/NonConstantFormat/nested.cpp

```diff
@@ -0,0 +1,81 @@
+typedef void *va_list;
+#define va_start(ap, parmN)
+#define va_end(ap)
+#define va_arg(ap, type) ((type)0)
+#define NULL 0
+
+extern "C" int printf(const char *fmt, ...);
+extern "C" int snprint(char *buf, int len, const char *fmt, ...);
+extern "C" int _vsnprintf_s(
+   char *buffer,
+   int sizeOfBuffer,
+   int count,
+   const char *fmt,
+   va_list argptr
+);
+extern "C" int snprintf ( char * s, int n, const char * format, ... );
+
+struct A {
+  void do_print(const char *fmt0) {
+    char buf[32];
+    snprintf(buf, 32, fmt0);
+  }
+};
+
+struct B {
+  A a;
+  void do_printing(const char *fmt) {
+    a.do_print(fmt);
+  }
+};
+
+struct C {
+  B b;
+  void do_some_printing(const char *fmt) {
+    b.do_printing(fmt);
+  }
+  const char *ext_fmt_str(void);
+};
+
+void foo(void) {
+  C c;
+  c.do_some_printing(c.ext_fmt_str());
```

Comment:
`// BAD [detected in the wrong location]`?

Reply:
Yes, I agree, this should be treated as BAD